pkg/mpc-examples/contracts/00-null-controller/NullControllerFactory.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// TODO license
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+// Core Infra
+import "./NullController.sol";
+import "../interfaces/IManagedPoolFactory.sol";
+import "@balancer-labs/v2-interfaces/contracts/vault/IVault.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/IManagedPool.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Create2.sol";
+
+/**
+ * @title NullControllerFactory
+ * @notice Factory for a Managed Pool and NullController.
+ * @dev Determines controller deployment address, deploys pool (w/ controller address as argument), then controller.
+ */
+contract NullControllerFactory {
+    mapping(address => bool) public isControllerFromFactory;
+
+    address public managedPoolFactory;
+    IVault public balancerVault;
+    bool public isDisabled;
+
+    uint256 private _nextControllerSalt;
+    address private _lastCreatedPool;
+
+    event ControllerCreated(address indexed controller, IVault vault, bytes32 poolId);
```

Comment:
The Vault will be the same for all controllers created out of this factory, so that argument seems a bit redundant.

Reply:
gone